src/main/java/com/entropyteam/entropay/auth/TokenUtils.java

```diff
@@ -0,0 +1,21 @@
+package com.entropyteam.entropay.auth;
+
+import java.util.Collection;
+import java.util.Map;
+import com.entropyteam.entropay.common.exceptions.AuthException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+
+public class TokenUtils {
+
+    public static final String COGNITO_CLAIMS_ROLE_KEY = "custom:role";
+
+    public static Map<String, Collection<String>> getRoles(Map<String, Object> tokenClaims) {
+        try {
+            return new ObjectMapper().readValue((String) tokenClaims.get(COGNITO_CLAIMS_ROLE_KEY), Map.class);
```

Comment:
afaik constructing the `ObjectMapper` is a relatively expensive operation, so creating one object and reusing it is recommended.
You could create a private final instance, like `private static final ObjectMapper objectMapper = new ObjectMapper()`

Personally, I wouldn't say I like utility classes, this guy explains it really well: https://www.jmgundersen.net/blog/utility-and-helper-classes-a-code-smell. It will be good if we can just put this code where it's actually needed or create a `@Component` and let spring handle the DI